pkg/kubelet/eviction/types.go

```diff
@@ -107,6 +108,11 @@ type Manager interface {
 
 	// IsUnderDiskPressure returns true if the node is under disk pressure.
 	IsUnderDiskPressure() bool
+
+	// PreemptyPods takes a list of insufficient resources, and attempts to free them by evicting pods
+	// based on requests.  For example, if the insufficient resource is 200Mb of memory, this function could
+	// evict a pod with request=250Mb.
+	PreemptPods(insufficientResources []predicates.InsufficientResourceError) error
```

Comment:
Did you consider making preemptions a separate module? Like admission controller for example?